sale_stock_available_to_promise_release/models/sale_order_line.py

```diff
@@ -23,13 +23,31 @@ class SaleOrderLine(models.Model):
         digits="Product Unit of Measure", compute="_compute_availability_status"
     )
 
+    def _on_order_route(self):
+        self.ensure_one()
+        mto_route = self.env.ref("stock.route_warehouse0_mto")
+        product_is_mto = mto_route in self.product_id.route_ids
+        line_is_mto = mto_route == self.route_id
+        return product_is_mto or line_is_mto
+
     @api.depends(
         "move_ids.ordered_available_to_promise_uom_qty",
         "product_id.route_ids",
+        "route_id",
         "product_uom_qty",
     )
     def _compute_availability_status(self):
         for record in self:
+            if record.display_type or not record.product_id:
+                record.availability_status = False
+                record.expected_availability_date = False
+                record.available_qty = False
+                continue
+            elif record.is_delivery:
```

Comment:
`is_delivery` and `display_type` should be added in `api.depends` to be consistent (they are not going to change over time that said)

Reply:
@mmequignon ping. Also, squash when done pls

Reply:
@mmequignon ping